src/System.Diagnostics.Process/tests/ProcessTests.Unix.cs

```diff
@@ -64,40 +63,57 @@ public void TestRootGetProcessById()
             Assert.Equal(1, p.Id);
         }
 
-        [Theory, InlineData(false), InlineData(true)] // Expected behavior varies on Windows and Unix. Refer to #23969
-        public void ProcessStart_TryOpenFolder_ThrowsWin32Exception(bool useShellExecute)
-        {
-            Win32Exception e = Assert.Throws<Win32Exception>(() => Process.Start(new ProcessStartInfo { UseShellExecute = useShellExecute, FileName = Path.GetTempPath() }));
-        }
-
-        [Fact, PlatformSpecific(TestPlatforms.Linux)]
+        [Theory, InlineData(true), InlineData(false)] // Expected behavior varies on Windows and Unix. Refer to #23969
         [OuterLoop("Opens program")]
-        public void ProcessStart_UseShellExecuteTrue_OpenFile_ThrowsIfNoDefaultProgramInstalledSucceedsOtherwise()
+        public void ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled(bool isFolder)
         {
-            string fileToOpen = GetTestFilePath() + ".txt";
-            File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecuteTrue_OpenFile_ThrowsIfNoDefaultProgramInstalledSucceedsOtherwise)}");
+            string fileToOpen;
+            string programToOpen = null;
+            if (isFolder)
+            {
+                fileToOpen = Environment.CurrentDirectory;
+            }
+            else
+            {
+                fileToOpen = GetTestFilePath() + ".txt";
+                File.WriteAllText(fileToOpen, $"{nameof(ProcessStart_UseShellExecute_OnUnix_ThrowsIfNoProgramInstalled)}");
+            }
 
-            string[] allowedProgramsToRun = { s_xdg_open, "gnome-open", "kfmclient" };
-            foreach (var program in allowedProgramsToRun)
+            if (RuntimeInformation.IsOSPlatform(OSPlatform.Linux))
             {
-                if (IsProgramInstalled(program))
+                string[] allowedProgramsToRun = { "xdg-open", "gnome-open", "kfmclient" };
+                foreach (var program in allowedProgramsToRun)
                 {
-                    var startInfo = new ProcessStartInfo { UseShellExecute = true, FileName = fileToOpen };
-                    using (var px = Process.Start(startInfo))
+                    if (IsProgramInstalled(program))
                     {
-                        Assert.NotNull(px);
-                        Console.WriteLine($"{nameof(ProcessStart_UseShellExecuteTrue_OpenFile_ThrowsIfNoDefaultProgramInstalledSucceedsOtherwise)}(): {program} was used to open file on this machine. ProcessName: {px.ProcessName}");
-                        Assert.Equal(program, px.ProcessName);
-                        px.Kill();
-                        px.WaitForExit();
-                        Assert.True(px.HasExited);
-                        Assert.Equal(s_exit_code_kill, px.ExitCode);
+                        programToOpen = program;
+                        break;
                     }
-                    return;
+                }
+                if (string.IsNullOrEmpty(programToOpen))
```

Comment:
Nit, as a matter of style and efficiency, we avoid IsNullOrEmpty as a "catch all check" ie., only if the value might actually be null or empty. In this case, it cannot be empty so you just need check for null.